### DISCUSSION TITLE
Why isn't the new findUniqueOrThrow error part of PrismaClientKnownRequestError?

### DISCUSSION THREAD

**Discussion by hymair** [prisma/prisma#14234] [Q&A] [answered]
The newly added methods are awesome but why are they throwing the generic `NotFoundError` which simply extends the default Error object?

code here: https://github.com/prisma/prisma/blob/main/packages/client/src/runtime/core/model/applyClientOnlyWrapper.ts#L34

Why not make them part of the `PrismaClientKnownRequestError`s?

So for example in this simple scenario we are checking by error.name: 
```js
try {
  return prisma.model.findUniqueOrThrow({ ... })
} catch (e) {
  if (e.name === 'NotFoundError') {
    ...
  }
}
```

Instead of checking like in all the other methods like this:
```js
try {
  return prisma.model.findUniqueOrThrow({ ... })
} catch (e) {
  if (e instanceof Prisma.PrismaClientKnownRequestError) {
    switch (e.code) {
      case 'P2002':
        throw new ConflictException('...')
      case 'P2025':
        throw new NotFoundException('...')
    }
  } else {
    throw new Error(e)
  }
}
```

**Comment by janpio** [comment, marked as answer]
See https://github.com/prisma/prisma/pull/14076

**Reply by hymair** [reply]
That's of course a better solution than checking by string. But my original question still stands. Shouldn't `findXOrThrow` throw a `PrismaClientKnownRequestError` with code 'P2025'?

**Reply by pantharshit00** [reply]
Hello! Sorry for late follow up here. 

I think you should open a feature request about this. You can open it here: https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=

I do agree having different error messages is not adequate and this can be improved so please open a feature request so this can be properly tracked.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
